pkg/f1/builder.go

```diff
@@ -0,0 +1,60 @@
+package f1
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/form3tech-oss/f1/pkg/f1/testing"
+)
+
+type F1 struct {
+	scenarios *testing.Scenarios
+	profiling *profiling
+}
+
+type profiling struct {
+	cpuProfile *os.File
+	memProfile string
+}
+
+func New() *F1 {
+	return &F1{
+		scenarios: testing.New(),
+		profiling: &profiling{},
+	}
+}
+
+func (f *F1) WithScenario(name string, setupFn testing.SetupFn) *F1 {
+	f.scenarios.AddByName(name, setupFn)
+	return f
+}
+
+func (f *F1) WithScenarioDescription(info testing.ScenarioInfo, setupFn testing.SetupFn) *F1 {
```

Comment:
wonder if we could change these to a single withScenario(name, setupfn, options...) - using options for the optional things like description, parameters, etc?

Reply:
Done, please take another look. We've changed it to this: `Scenarios().Add(name, fn, options...)...Execute()`